text/003-trigger-authority.md

```diff
@@ -0,0 +1,134 @@
+# Refine Triggers Authority Mechanism
+
+## Context
+
+Every trigger is composed of:
+
+- An executable - a set of actions to execute
+- An **authority** - an account ID, from under which trigger's actions are executed.
+- A filter (or the _kind_ of trigger) - to define a condition where to execute the trigger, one of:
+  - Pipeline - to react on blocks/transactions
+  - Time - to execute at a certain time, either scheduled or pre-commit
+  - Trigger completed - to handle completion of other triggers
+  - Data - to react on state changes (e.g. domain registered)
+  - _Execute trigger_ filter, or _by-call_ filter[^1] - states that this trigger is not automatically executed, but only
+    via `ExecuteTrigger` instruction.
+
+[^1]: By-call triggers are subject for removal in #5147
+
+Currently, the **authority** of the trigger behaves differently depending on the kind of trigger (i.e. on its filter):
+
+| Trigger Kind                                                               | Trigger execution authority                                  |
+| -------------------------------------------------------------------------- | ------------------------------------------------------------ |
+| Pipeline                                                                   | Use trigger's                                                |
+| Time                                                                       | Use trigger's                                                |
+| Trigger completed                                                          | Use trigger's                                                |
+| Data                                                                       | Use caller's (the one who caused the data event)             |
+| By-call                                                                    | Use caller's (who executed the `ExecuteTrigger` instruction) |
+| Also, if any of the data triggers fail, the entire transaction also fails. |                                                              |
```

Comment:
In my view, the only clearly useful programmable execution systems currently identified are the following three:

* smart contract (integrated with by-call trigger and custom instruction)
* data trigger
* time trigger

I am skeptical about providing other types of triggers. For details, see https://github.com/hyperledger-iroha/iroha/issues/5512.